src/smith/differentiable_numerics/lumped_mass_explicit_newmark_state_advancer.hpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Lawrence Livermore National Security, LLC and
+// other Smith Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file lumped_mass_explicit_newmark_state_advancer.hpp
+ *
+ * @brief Implementation of explicit Newmark
+ */
+
+#pragma once
+
+#include <vector>
+#include "gretl/double_state.hpp"
+#include "smith/physics/common.hpp"
+#include "smith/differentiable_numerics/field_state.hpp"
+#include "smith/differentiable_numerics/state_advancer.hpp"
+
+namespace smith {
+
+class WeakForm;
+class TimestepEstimator;
+class BoundaryConditionManager;
+
+/// Lumped mass explicit dynamics implementation for the StateAdvancer interface
+class LumpedMassExplicitNewmarkStateAdvancer : public StateAdvancer {
```

Comment:
Does it make sense to (or is it even possible to) separate the time integration logic from the differentiation logic?